lib/redmine_s3/attachment_patch.rb

```diff
@@ -30,12 +31,44 @@ def put_to_s3
       end
 
       def delete_from_s3
-        logger.debug("Deleting #{disk_filename}")
-        RedmineS3::Connection.delete(disk_filename)
+        logger.debug("Deleting #{disk_filename_s3}")
+        RedmineS3::Connection.delete(disk_filename_s3)
       end
 
       # Prevent file uploading to the file system to avoid change file name
       def files_to_final_location; end
+
+      # Returns the full path the attachment thumbnail, or nil
+      # if the thumbnail cannot be generated.
+      def thumbnail_s3(options={})
+        return nil unless Object.const_defined?(:Magick)
```

Comment:
Please, do not return `nil` explicitly. It doesn't make sense. You even don't use the result of the method - so there is no reason to return something here explicitly. Just return will be enough.